### PR TITLE
Ensure search box has focus when SR dialog popped

### PR DESCRIPTION
**Pull request by windymilla**
See issue #373 for detailed description of problem

### CODE REVIEW

**Comment by windymilla** [comment]
Testing notes: you are trying to engineer a situation where the S/R dialog is showing, and the Search field is "selected" (highlighted), but because the focus is not in the Search field (e.g. the Replace field or nowhere), when you press Cmd+C to copy the selected text, you don't get it.

There are lots of ways to do this, but one short way is to use WF, All Words. 
Then Cmd+mouse-1 on a word in the WF list. 
The S/R dialog should pop with the word selected in the Search field.
The focus **is** in Search field, so if you do Cmd+C, it copies the word into your copy/paste buffer. 
Then, without dismissing the S/R dialog, do Cmd+mouse-1 again in the WF dialog. 
This time, the word is selected in the Search field, but the focus (and insert cursor) are not.
Cmd+C now will not copy the apparently selected word from the Search field.
In this PR, the focus should be forced to the search field each time a word in the Search field is selected, so Cmd+C will behave as expected.

**Comment by srjfoo** [comment]
I tried the testing notes, but `master`, another random branch, and `sr-dialog` all behaved the same:

* If GG2 was in the foreground, cmd-left-click on the WF word would fill in the word in the sr dialog, and select it. However, note that it doesn't move the highlighting to any subsequent word after the first one that is clicked without the cmd key being used in conjunction. That is if I left-click on "Amphidolops", and then cmd-left-click on "Anacodon", "Anacodon" is highlighted in s/r search field, and "Amphidolops" remains highlighted in the main text window (whether or not I had caused it to be inserted in the search field in the sr dialog).
* I discovered, however, that if I was using the text input field for github comments as a storage place for the selected words, and forgot to force the active app to be GG2 (i.e. Python), but cmd-left-clicked on words in the WF list, it would cheerfully fill in the search field in the SR dialog, and cheerfully move the highlighted word in WF, but would not bring Python/GG2 to the foreground, would not change the highlighting in the main text window, and would not highlight the word in the search field in the SR dialog. (I hope all this makes sense.)

Short version -- with this particular test, if I'm doing it right, I saw no difference among the three branches I tested in.

**Review by srjfoo — APPROVED** [review]
